proposed/logging-generator.md

```diff
@@ -0,0 +1,621 @@
+
+There is no constraint in the ordering at this point. So the user could define `ILogger` as the last argument for example:
+```csharp
+[LoggerMessage(110, LogLevel.Debug, "M1 {ex3} {ex2}")]
+static partial void LogMethod(System.Exception ex, System.Exception ex2, System.Exception ex3, ILogger logger);
+```
+
+The samples below show we could:
+
+- `LogWithCustomEventName`: retrieve event name via `LoggerMessage` attribute. 
+- `LogWithDynamicLogLevel`: set log level dynamically, to allow log level to be set based on configuration input.
+- `UsingFormatSpecifier`: use format specifiers to format logging parameters.
+
+```csharp
+public partial class LoggingSample5
+{
+    private readonly ILogger _logger;
+
+    public LoggingSample5(ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    [LoggerMessage(20, LogLevel.Critical, "Value is {value:E}")]
+    public static partial void UsingFormatSpecifier(ILogger logger, double value);
+
+    [LoggerMessage(9, LogLevel.Trace, "Fixed message", EventName = "CustomEventName")]
+    public partial void LogWithCustomEventName();
+
+    [LoggerMessage(10, "Welcome to {city} {province}!")]
+    public partial void LogWithDynamicLogLevel(string city, LogLevel level, string province);
+
+    public void TestLogging()
+    {
+        LogWithCustomEventName();
+        LogWithDynamicLogLevel("Vancouver", LogLevel.Warning, "BC");
+        LogWithDynamicLogLevel("Vancouver", LogLevel.Information, "BC");
+        double val = 12345.6789;
+        Log.UsingFormatSpecifier(logger, val);
+    }
+}
+```
+output to `TestLogging()` using SimpleConsole:
+```
+trce: LoggingExample[9]
+      Fixed message
+warn: LoggingExample[10]
+      Welcome to Vancouver BC!
+info: LoggingExample[10]
+      Welcome to Vancouver BC!
+crit: LoggingExample[20]
+      Value is 1.234568E+004
+```
+
+same console logs formatted using JsonConsole:
+```
+{
+  "EventId": 9,
+  "LogLevel": "Trace",
+  "Category": "LoggingExample",
+  "Message": "Fixed message",
+  "State": {
+    "Message": "Fixed message",
+    "{OriginalFormat}": "Fixed message"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Warning",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Information",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 20,
+  "LogLevel": "Critical",
+  "Category": "LoggingExample",
+  "Message": "Value is 1.234568E+004",
+  "State": {
+    "Message": "Value is 1.234568E+004",
+    "value": 12345.6789,
+    "{OriginalFormat}": "Value is {value:E}"
+  }
+}
+```
+
+
+#### `EmitDefaultMessage` : YES/NO
```

Comment:
I chatted with @shirhatti and @davidfowl offline about this feature and I didn't get an up vote for keeping this feature.

I'll remove it from the proposal since 
(A) It is not an already supported scenario.
(B) The behavior as seen from the console output seems a bit unexpected. 

cc: @geeknoid